Userland/Libraries/LibWeb/Layout/BlockFormattingContext.h

```diff
@@ -150,6 +150,28 @@ class BlockFormattingContext : public FormattingContext {
         }
     };
 
+    Vector<CSSPixels> y_offset_state_stack;
```

Comment:
Why does this need to be a stack? When do we have more than one Y offset?

Reply:
> Why does this need to be a stack?

I did that just to have current y in BFC members so it doesn't have to be explicitly passed to methods that need it.
Turned out to make clearance work correctly `place_block_level_element_in_normal_flow_vertically` needs to have both: 1) y offset without clearance adjustment to place a current box and 2) reference to y offset to adjust it with clearance so the next block is placed correctly. so when current y offset is stored in BFC members only 2) need to be passed as a parameter.

> When do we have more than one Y offset?

every block container has it's own y offset. but there is only one y offset at a time hence it's stack. maybe I got this question wrong, sorry.

Reply:
I see that making an explicit stack just to have it in class members is a bit an overkill. I can redo that.

Reply:
How would you do it differently?

Reply:
I went with just pointer to current y offset stored in BFC member.